app/logic/Mail/ActiveSync/ActiveSyncFolder.ts

```diff
@@ -174,6 +174,17 @@ export class ActiveSyncFolder extends Folder implements ActiveSyncPingable {
     });
     this.messages.addAll(newMsgs);
     this.account.addPingable(this);
+    if (this.specialFolder == SpecialFolder.Sent) {
```

Comment:
Here, add a possibility to add an observer function.
The sending code then adds an observer and does what you do here.